spec/models/missing_link_finder_spec.rb

```diff
@@ -174,5 +176,38 @@
         error: "#{openssl_error.class}: #{openssl_error.message}"
       })
     end
+
+    context "when links are xlink:href" do
+      it "returns a good status when the page has both links" do
+        body = <<~BODY
+          <svg>
+            <a xlink:href="#{next_link}">next</a>
+            <a xlink:href="#{prev_link}">prev</a>
+          </svg>
+        BODY
+        stub_request(:get, redirection.url).to_return(body: body)
+
+        result = MissingLinkFinder.new(redirection).run
+
+        expect(result).to eq({status: :good})
+      end
+
+      %w[next prev].each do |type|
+        it "returns a bad status when the page has only #{type} link" do
+          link = {"next" => next_link, "prev" => prev_link}[type]
+          body = <<~BODY
+            <svg>
+              <a xlink:href="#{link}">#{type}</a>
+            </svg>
+          BODY
+          stub_request(:get, redirection.url).to_return(body: body)
+
+          result = MissingLinkFinder.new(redirection).run
+
+          missing = %w[next prev] - [type]
```

Comment:
At first I thought iterating over the types to make two tests easy would be fine, but this and the `link` variable above are starting to push into too much logic in the test itself I think. Up to you though if you want to rewrite it.

Reply:
I agree, it's a bit complex. I split it into two tests: 09006574e8710135e866f1ec747e3ca2a2c2d301